tests/e2e/kubernetes/kubernetes_test.go

```diff
@@ -203,6 +214,98 @@ func testInstall(haMode bool) func(t *testing.T) {
 	}
 }
 
+func testMtls(isDaprInstalled, enabled bool) func(t *testing.T) {
+	return func(t *testing.T) {
+		daprPath := getDaprPath()
+		output, err := spawn.Command(daprPath, "mtls", "-k")
+		if !isDaprInstalled {
+			require.Error(t, err, "expected error to be return if dapr not installed")
+			require.Contains(t, output, "error checking mTLS: system configuration not found", "expected output to match")
+			return
+		}
+		require.NoError(t, err, "expected no error on querying for mtls")
+		if !enabled {
+			require.Contains(t, output, "Mutual TLS is disabled in your Kubernetes cluster", "expected output to match")
+		} else {
+			require.Contains(t, output, "Mutual TLS is enabled in your Kubernetes cluster", "expected output to match")
+		}
+
+		//expiry
+		output, err = spawn.Command(daprPath, "mtls", "expiry")
+		require.NoError(t, err, "expected no error on querying for mtls expiry")
+		assert.Contains(t, output, "Root certificate expires in", "expected output to contain string")
+		assert.Contains(t, output, "Expiry date:", "expected output to contain string")
+
+		//export
+		// check that the dir does not exist now
+		_, err = os.Stat("./certs")
+		if assert.Error(t, err) {
+			assert.True(t, os.IsNotExist(err), err.Error())
+		}
+
+		output, err = spawn.Command(daprPath, "mtls", "export", "-o", "./certs")
+		require.NoError(t, err, "expected no error on mtls export")
+		require.Contains(t, output, "Trust certs successfully exported to", "expected output to contain string")
+
+		// check export success
+		_, err = os.Stat("./certs")
+		require.NoError(t, err, "expected directory to exist")
+		_, err = os.Stat("./certs/ca.crt")
+		require.NoError(t, err, "expected file to exist")
+		_, err = os.Stat("./certs/issuer.crt")
+		require.NoError(t, err, "expected file to exist")
+		_, err = os.Stat("./certs/issuer.key")
+		require.NoError(t, err, "expected file to exist")
+		err = os.RemoveAll("./certs")
+		if err != nil {
+			t.Logf("error on removing local certs directory %s", err.Error())
+		}
+	}
+}
+
+func testComponents(isDaprInstalled bool) func(t *testing.T) {
```

Comment:
nit: This construction is a little weird to me. The isDaprInstalled case and the !isDaprInstalled case don't actually share any code? Doesn't it make things a little clearer if they were just separate functions?

Reply:
Will be incorporating those changes in #665 . Already fixed it as part of that code.....